pkg/cmd/experimental/syncgroups/sync-groups.go

```diff
@@ -0,0 +1,188 @@
+package syncgroups
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/util"
+	kerrs "k8s.io/kubernetes/pkg/util/errors"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+const (
+	SyncGroupsRecommendedName = "sync-groups"
+
+	syncGroupsLong = `
+Sync OpenShift Groups with records from an external provider.
+
+`
+	syncGroupsExamples = `  // Sync all groups from an LDAP server
+  $ %[1]s ldap all --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync specific groups specified in a whitelist file with an LDAP server 
+  $ %[1]s ldap whitelist /path/to/whitelist.txt --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync all OpenShift Groups that have already been synced with an LDAP server
+  $ %[1]s openshift all --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync specific OpenShift Groups if they have already been synced with an LDAP server
+  $ %[1]s openshift whitelist groups/group1 groups/group2 groups/group3 --sync-config=/path/to/ldap-sync-config.yaml
+`
+)
+
+// GroupSyncScope determines the scope of the group sync operation
+type GroupSyncScope string
+
+const (
+	// GroupSyncScopeAll determines that all groups from the source should be synced
+	GroupSyncScopeAll GroupSyncScope = "All"
+	// GroupSyncScopeWhitelist determines that a whitelist of groups from the source should be synced
+	GroupSyncScopeWhitelist GroupSyncScope = "Whitelist"
+)
+
+func ValidateScope(scope GroupSyncScope) bool {
+	knownScopes := util.NewStringSet(string(GroupSyncScopeAll), string(GroupSyncScopeWhitelist))
+	return knownScopes.Has(string(scope))
+}
+
+// GroupSyncSource determines the source of the groups to be synced
+type GroupSyncSource string
+
+const (
+	// GroupSyncSourceLDAP determines that the groups to be synced are determined from an LDAP record
+	GroupSyncSourceLDAP GroupSyncSource = "LDAP"
+	// GroupSyncSourceOpenShift determines that the groups to be synced are determined from OpenShift records
+	GroupSyncSourceOpenShift GroupSyncSource = "OpenShift"
+)
+
+func ValidateSource(source GroupSyncSource) bool {
+	knownSources := util.NewStringSet(string(GroupSyncSourceLDAP), string(GroupSyncSourceOpenShift))
+	return knownSources.Has(string(source))
+}
+
+type SyncGroupsOptions struct {
```

Comment:
I'm going to keep *this* struct parallel with those in other commands - the `Options` contains options given as input to the command on the command line.